src/mbgl/style/layers/symbol_layer.cpp

```diff
@@ -125,7 +125,7 @@ void SymbolLayer::setIconOptional(PropertyValue<bool> value) {
     impl->layout.iconOptional.set(value);
 }
 PropertyValue<AlignmentType> SymbolLayer::getDefaultIconRotationAlignment() {
-    return { AlignmentType::Viewport };
+    return { AlignmentType::Undefined };
```

Comment:
Looks like this will cause the SDKs to potentially return a bogus value for these properties by default. `AlignmentType::Undefined` has no analogue in the style specification, so [`MGLSymbolStyleLayerTextRotationAlignment`](https://github.com/mapbox/mapbox-gl-native/blob/7e208c46cebe5b49d4b7ead9826bbe4229ebabd3/platform/darwin/src/MGLSymbolStyleLayer.h#L31) for instance has no corresponding value. On iOS and macOS, this means `-[MGLSymbolStyleLayer textRotationAlignment]` will return a value of 3 that isn’t describable as an `MGLSymbolStyleLayerTextRotationAlignment`.

These default getters were added in order for the iOS and macOS SDKs to implement `null_resettable` semantics (#6126). It would be unfortunate if these SDKs had to special-case these two properties, but I’m not sure how else to avoid having yet another layer of getters for these style properties.

/cc @frederoni